doc/user/content/integrations/http-api.md

```diff
@@ -0,0 +1,194 @@
+      statements, e.g. `SELECT 1; SELECT 2;`
+    - Treats all statements as in an implicit transaction unless other
+      transaction control is invoked.
+- **Extended**, which mirrors PostgreSQL's [Extended Query][extended-query] protocol.
+    - Supports multiple queries, but only one statement per query string.
+    - Supports parameters.
+    - Eagerly commits DDL (e.g. `CREATE TABLE`) in implicit transactions, but
+      not DML (e.g. `INSERT`).
+
+## Usage
+
+### Endpoint
+
+```
+https://<MZ host address>/api/sql
+```
+
+Accessing the endpoint requires username + password authentication.
+
+### Input format
+
+#### Simple
+
+The request body is a JSON object containing a key, `query`, which specifies the
+SQL string to execute. `query` may contain multiple SQL statements separated by
+semicolons.
+
+```json
+{
+    "query": "select * from a; select * from b;"
+}
+```
+
+#### Extended
+
+The request body is a JSON object containing a key `queries`, whose value is
+array of objects, whose structure is:
+
+Key | Value
+----|------
+`query` | A SQL string containing one statement to execute
+`params` | An optional array of text values to be used as the parameters to `query`. _null_ values are converted to _null_ values in Materialize. Note that all parameter values' elements must be text or _null_; the API will not accept JSON numbers.
+
+```json
+{
+    "queries": [
+        { "query": "select * from a;" },
+        { "query": "select a + $1 from a;", "params": ["100"] }
+        { "query": "select a + $1 from a;", "params": [null] }
+    ]
+}
+```
+
+#### TypeScript definition
+
+You can model these with the following TypeScript definitions:
+
+```typescript
+interface Simple {
+    query: string;
+}
+
+interface ExtendedRequest {
+    query: string;
+    params?: (string | null)[];
+}
+
+interface Extended {
+    queries: ExtendedRequest;
+}
+
+type Sql = Simple | Extended;
+```
+
+### Output format
+
+The output format is a JSON object with one key, `results`, whose value is
+an array of the following:
+
+The serialized `environmentd::http::SimpleResult` values are represented in JSON as:
+
+Result | JSON value
+---------------------|------------
+Rows | `{"rows": <2D array of JSON-ified results>, "col_names": <array of text>}`
+Error | `{"error": <Error string from execution>}`
+Ok | `{ "ok": <tag>}`
+
+Each committed statement returns exactly one of these values; e.g. in the case
+of "complex responses", such as `INSERT INTO...RETURNING`, the presence of a
+`"rows"` object implies `"ok"`.
+
+Note that the returned values include the results of statements which were
+ultimately rolled back because of an error in a later part of the transaction.
+Users must parse the results to understand which statements ultimately reflect
+the resultant state.
+
+## Examples
+
+```bash
+curl https://`https://<MZ host address>/api/sql/api/sql \
+    -H "Content-Type: application/json" \
+    -u <username>:<passsword> \
+    -d '{
```

Comment:
Super minor, but I like using long options in CLI examples since they're a bit more self documenting:

```suggestion
    --header "Content-Type: application/json" \
    --user <username>:<passsword> \
    --data '{
```